### PR TITLE
Service cleanup

### PR DESCRIPTION
**Pull request by sharwell**
Review commit-by-commit is likely to minimize complexity.

There are many more cases where `IVsService<>` can be used to simplify code, but I wanted to send the changes in smaller reviews to keep reviews as simple as possible.

### CODE REVIEW

File: src/VisualStudio/Core/Def/RoslynPackage.cs [hunk 1]
```diff
@@ -218,7 +218,7 @@ protected override async Task LoadComponentsAsync(CancellationToken cancellation
 
             // we need to load it as early as possible since we can have errors from
             // package from each language very early
-            await this.ComponentModel.GetService<TaskCenterSolutionAnalysisProgressReporter>().InitializeAsync(this).ConfigureAwait(false);
+            await this.ComponentModel.GetService<TaskCenterSolutionAnalysisProgressReporter>().InitializeAsync().ConfigureAwait(false);
```
Comment (Cosifne) [inline src/VisualStudio/Core/Def/RoslynPackage.cs:221]: Why `InitializeAsync()` instead of `initializeAsync(this)`?

**Review by Cosifne — APPROVED** [review]
One minor question

File: src/VisualStudio/Core/Def/RoslynPackage.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/Core/Def/RoslynPackage.cs:221]: ➡️ `this` was only passed so the target would have access to a service provider. It now gets the service provider from MEF during construction.

File: src/VisualStudio/Core/Def/ColorSchemes/ColorSchemeApplier.cs [hunk 2]
```diff
@@ -43,6 +45,7 @@ internal sealed partial class ColorSchemeApplier
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public ColorSchemeApplier(
             IThreadingContext threadingContext,
+            IVsService<SVsFontAndColorStorage, IVsFontAndColorStorage> fontAndColorStorage,
             IGlobalOptionService globalOptions,
             SVsServiceProvider serviceProvider)
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/ColorSchemes/ColorSchemeApplier.cs:50]: Was going to leave this one for another PR?

File: src/VisualStudio/Core/Def/ErrorReporting/VisualStudioErrorReportingService.cs [hunk 3]
```diff
@@ -19,6 +20,7 @@ namespace Microsoft.CodeAnalysis.ErrorReporting
     internal partial class VisualStudioErrorReportingService : IErrorReportingService
     {
         private readonly IThreadingContext _threadingContext;
+        private readonly IVsService<IVsActivityLog> _activityLog;
         private readonly IAsynchronousOperationListener _listener;
         private readonly VisualStudioInfoBar _infoBar;
         private readonly SVsServiceProvider _serviceProvider;
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/ErrorReporting/VisualStudioErrorReportingService.cs:26]: This appears to be unused now, but could be wrong if GitHub is making it difficult for me.

File: src/VisualStudio/Core/Def/Options/VisualStudioOptionPersisterProvider.cs [hunk 4]
```diff
@@ -50,11 +49,27 @@ internal sealed class VisualStudioOptionPersisterProvider : IOptionPersisterProv
...
-                    new FeatureFlagPersister(await TryGetServiceAsync(_featureFlagsService).ConfigureAwait(false)));
+        {
+            return _lazyPersister ??= await CreatePersisterAsync(cancellationToken).ConfigureAwait(true);
+
+            async Task<VisualStudioOptionPersister> CreatePersisterAsync(CancellationToken cancellationToken)
+            {
+                // Obtain services before creating instances. This avoids state corruption in the event cancellation is
+                // requested (some of the constructors register event handlers that could leak if cancellation occurred
+                // in the middle of construction).
+                var settingsManager = await _settingsManagerService.GetValueAsync(cancellationToken).ConfigureAwait(true);
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Options/VisualStudioOptionPersisterProvider.cs:60]: Should we just not have cancellation here? I can't imagine we're really passing a meaningful cancellation token here.

**Review by jasonmalinowski — APPROVED** [review]
Nice! If nothing else, it cleans up a lot of our inconsistencies around fetching some of these services that have crept in over the years.

File: src/VisualStudio/Core/Def/Options/VisualStudioOptionPersisterProvider.cs (hunk 4 above)
Comment (sharwell) [reply src/VisualStudio/Core/Def/Options/VisualStudioOptionPersisterProvider.cs:60]: ➡️ It can happen if VS closes shortly after launch

**Comment by sharwell** [comment]
> Nice! If nothing else, it cleans up a lot of our inconsistencies around fetching some of these services that have crept in over the years.

Yes. This is only a few of the total set of changes needed to move to `IVsService<>`, but it definitely makes things cleaner and more consistent.

File: src/VisualStudio/Core/Def/ErrorReporting/VisualStudioErrorReportingService.cs (hunk 3 above)
Comment (sharwell) [reply src/VisualStudio/Core/Def/ErrorReporting/VisualStudioErrorReportingService.cs]: ➡️ Amended the change to remove this unused field.
